iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/entity/AbstractEntityExporterProcess.java

```diff
@@ -43,225 +41,219 @@
-		} finally {
-			it.close();
-			actionManager.close();
-		}
-	}
-	
-	/**
-	 * Handles single MDStore record.
-	 * @param mdStoreRecord
-	 * @param actionSetId
-	 * @param actionFactory
-	 * @param actionManager
-	 * @throws Exception 
-	 */
-	protected void handleRecord(String mdStoreRecord, 
-			String actionSetId, ActionFactory actionFactory, 
-			ActionManagerServiceFacade actionManager) throws Exception {
-		if (mdStoreRecord!=null) {
-			XsltInfoPackageAction xsltAction = actionFactory.generateInfoPackageAction(
-					entityXSLTName, actionSetId, 
-					configProvider.provideAgent(), 
-					Operation.INSERT, mdStoreRecord,
-					configProvider.provideProvenance(),
-					configProvider.provideNamespacePrefix(),
-					configProvider.provideActionTrust());
-			if (xsltAction!=null) {
-				List<AtomicAction> atomicActions = xsltAction.asAtomicActions();
-				if (atomicActions!=null) {
-					actionManager.storeAction(atomicActions,
-							configProvider.provideProvenance(),
-							configProvider.provideActionTrust(),
-							configProvider.provideNamespacePrefix());	
-				}
-			}
-		}
-	}
-	
-	/**
-	 * Provides action manager instance.
-	 * @param conf
-	 * @param parameters
-	 * @return action manager instance
-	 * @throws IOException
-	 */
-	protected ActionManagerServiceFacade buildActionManager(Configuration conf, Map<String, String> parameters)
+    private static final String MDSTORE_FACADE_FACTORY_CLASS = "mdstore.facade.factory.classname";
+
+    private static final Provenance PROVENANCE_DEFAULT = Provenance.sysimport_mining_repository;
+
+    private final Logger log = Logger.getLogger(this.getClass());
+
+    private final static String inputPort = "input";
+
+    private final Schema inputPortSchema;
+
+    private final String entityXSLTName;
+
+    private final String entityXSLTLocation;
+
+    private final String entityNamespacePrefix;
+    
+    private final ActionFactory actionFactory;
+
+    // ------------------------ CONSTRUCTORS -----------------------------
+
+    /**
+     * @param inputPortSchema input port avro schema
+     * @param entityXSLTName entity XSL transformation name
+     * @param entityXSLTLocation entity XSL transformation location
+     * @param entityNamespacePrefix namespace prefix to be used with given entity type
+     */
+    public AbstractEntityExporterProcess(Schema inputPortSchema, String entityXSLTName, String entityXSLTLocation,
+            String entityNamespacePrefix) {
+        this.inputPortSchema = inputPortSchema;
+        this.entityXSLTName = entityXSLTName;
+        this.entityXSLTLocation = entityXSLTLocation;
+        this.entityNamespacePrefix = entityNamespacePrefix;
+        this.actionFactory = buildActionFactory();
+
+    }
+
+    // ------------------------ LOGIC -----------------------------
+
+    @Override
+    public void run(PortBindings portBindings, Configuration conf, Map<String, String> parameters) throws Exception {
```

Comment:
It would be great to have some description of the method here. Mind, the method signature is very generic.